### PR TITLE
Update local ssh troubleshooting

### PR DESCRIPTION
**Pull request by jeanp413**

### CODE REVIEW

File: src/content/docs/references/ides-and-editors/vscode.md [hunk 1]
```diff
@@ -221,4 +221,6 @@ If you are still having issues connecting to VS Code Desktop from Gitpod, try:
 
 1. Check if your VS Code Desktop application and Gitpod VS Code Desktop extension version are all using the latest version.
 2. Identify [which method you're using to connect to Gitpod](/docs/references/ides-and-editors/vscode#how-do-i-know-how-im-connecting-to-vs-code-desktop), if it's `SSH Gateway` or `Local Companion` see their troubleshooting [here](/docs/references/ides-and-editors/legacy-vscode-connections#troubleshooting)
-3. If it's Local SSH, please [export log](/docs/help/troubleshooting#gitpod-logs-in-vs-code-browser-and-desktop) on the failed to connect window and [email us](mailto:contact@gitpod.io)
+3. If it's Local SSH:
```
Comment (akosyakov) [inline src/content/docs/references/ides-and-editors/vscode.md:224]: We should not surface Local SSH in user docs, proxy is internal. @loujaybee Maybe you know how to word it better.

Comment (akosyakov) [reply src/content/docs/references/ides-and-editors/vscode.md:224]: It should be more like enable SSH connection debug logging.

File: src/content/docs/references/ides-and-editors/vscode.md [hunk 2]
```diff
@@ -221,4 +221,6 @@ If you are still having issues connecting to VS Code Desktop from Gitpod, try:
 
 1. Check if your VS Code Desktop application and Gitpod VS Code Desktop extension version are all using the latest version.
 2. Identify [which method you're using to connect to Gitpod](/docs/references/ides-and-editors/vscode#how-do-i-know-how-im-connecting-to-vs-code-desktop), if it's `SSH Gateway` or `Local Companion` see their troubleshooting [here](/docs/references/ides-and-editors/legacy-vscode-connections#troubleshooting)
```
Comment (akosyakov) [inline src/content/docs/references/ides-and-editors/vscode.md:223]: We should rewrite it completely SSH Gateway or Local Companion should be rather considered legacy.

**Comment by akosyakov** [comment]
I'm usually pointing to https://www.gitpod.io/docs/help/troubleshooting#gitpod-logs-in-vs-code-browser-and-desktop
I would prefer if a setting is surfaced there.

**Comment by jeanp413** [comment]
image
image

**Review by akosyakov — APPROVED** [review]
I refined it, we need to keep legacy docs for sometime, since users need time to upgrade to latest versions and SSH Gateway is stil relevant. 

I refined though to nudge users away from SSH Gateway.

File: src/content/docs/references/ides-and-editors/vscode.md [hunk 3]
```diff
@@ -217,8 +217,9 @@ You can upload files to your workspace by dragging and dropping them into the ed
 
 ## Troubleshooting
 
-If you are still having issues connecting to VS Code Desktop from Gitpod, try:
+If you are having issues connecting to VS Code Desktop from Gitpod, try:
```
Comment (filiptronicek) [inline src/content/docs/references/ides-and-editors/vscode.md:220]:
```suggestion
If you are having issues connecting to VS Code Desktop from Gitpod, try to:
```

File: src/content/docs/references/ides-and-editors/vscode.md [hunk 4]
```diff
@@ -217,8 +217,9 @@ You can upload files to your workspace by dragging and dropping them into the ed
...
-3. If it's Local SSH, please [export log](/docs/help/troubleshooting#gitpod-logs-in-vs-code-browser-and-desktop) on the failed to connect window and [email us](mailto:contact@gitpod.io)
+1. Ensure both your VS Code Desktop application and Gitpod VS Code Desktop extension are up-to-date.
+2. Determine [your current connection method to Gitpod](/docs/references/ides-and-editors/vscode#how-do-i-know-how-im-connecting-to-vs-code-desktop).
+3. If you're not using the **recommended** mode, [switch to it](/docs/references/ides-and-editors/vscode#connecting-to-vs-code-desktop). If you can't switch, consult the [legacy docs](/docs/references/ides-and-editors/legacy-vscode-connections#troubleshooting).
+4. If you're already using the **recommended** mode, [export the logs](/docs/help/troubleshooting#gitpod-logs-in-vs-code-browser-and-desktop) from the failed to connect window and [email us](mailto:contact@gitpod.io).
```
Comment (filiptronicek) [inline src/content/docs/references/ides-and-editors/vscode.md:225]:
```suggestion
3. [Switch to the recomennded mode](/docs/references/ides-and-editors/vscode#connecting-to-vs-code-desktop), if you aren't using it. If you can't switch, consult the [legacy docs](/docs/references/ides-and-editors/legacy-vscode-connections#troubleshooting).
4. [Export the logs](/docs/help/troubleshooting#gitpod-logs-in-vs-code-browser-and-desktop) from the failed to connect window and [email us](mailto:contact@gitpod.io) if you're already using the **recommended** mode.
```

File: src/content/docs/help/troubleshooting.md [hunk 5]
```diff
@@ -16,9 +16,24 @@ If you cannot find your issue here or in the documentation, please contact Gitpo
...
+**Caution**: Do **NOT** share these logs publicly as they may contain sensitive information. Send them to [troubleshooting@gitpod.io](mailto:troubleshooting@gitpod.io) and include a link to a relevant GitHub issue if needed.
+
+### VS Code Desktop: Workspace Connectivity Tracing
+
+If experiencing connectivity issues on VS Code Desktop, enable detailed logging:
+
+-   In your VS Code settings, set `gitpod.lssh.logLevel` to `debug`.
+-   Reproduce the issue, and after reestablishing the connection, collect logs from the affected window.
+
+![Enable local ssh debug logging](/images/editors/vscode-desktop-localssh-debug.png)
```
Comment (filiptronicek) [inline src/content/docs/help/troubleshooting.md:34]: We need to keep a clear distinction between local ssh (like the local command on a user's machine) and our feature, Local SSH.
```suggestion
![Enable Local SSH debug logging](/images/editors/vscode-desktop-localssh-debug.png)
```

Comment (akosyakov) [reply src/content/docs/help/troubleshooting.md:34]: it should be `Enable Workspace Connectivity Tracing`, we should not talk about Local SSH

File: src/content/docs/references/ides-and-editors/vscode.md (hunk 4 above)
Comment (akosyakov) [reply src/content/docs/references/ides-and-editors/vscode.md:225]: I think it reads better as before with conditional logic:
- not recommended to that
- recommended do that

File: src/content/docs/help/troubleshooting.md (hunk 5 above)
Comment (jeanp413) [reply src/content/docs/help/troubleshooting.md:34]:
```suggestion
![Enable Workspace Connectivity Tracing](/images/editors/vscode-desktop-localssh-debug.png)
```

**Pull request merged**